src/Consolonia.Core/Infrastructure/AnsiConsoleOutput.cs

```diff
@@ -151,33 +138,16 @@ public void WritePixel(PixelBufferCoordinate position, in Pixel pixel)
             }
             else
             {
-                // complex symbol, could be emoji or other multi-char glyph
-                if (SupportsEmojiVariation)
-                {
-                    SetCaretPosition(position);
-                    WriteText(pixel.Foreground.Symbol.Complex);
-                    position = new PixelBufferCoordinate((ushort)(position.X + pixel.Width), position.Y);
-                }
-                else
-                {
-                    // rendering over the top with the glyph.
-                    // process each glyph, rendering the width as spaces then moving the cursor and
-                    foreach (Grapheme grapheme in Grapheme.Parse(pixel.Foreground.Symbol.Complex, SupportsComplexEmoji))
-                    {
-                        ushort glyphWidth = grapheme.Glyph.MeasureText();
-                        if (glyphWidth > 1)
-                        {
-                            WriteText(Esc.SetCursorPosition(position.X + 1, position.Y));
-                            WriteText(new string(' ', Math.Min(Size.Width - position.X - 1, glyphWidth - 1)));
-                        }
-
-                        WriteText(Esc.SetCursorPosition(position.X, position.Y));
-                        WriteText(grapheme.Glyph);
-
-                        position =
-                            new PixelBufferCoordinate((ushort)(position.X + glyphWidth), position.Y);
-                    }
-                }
+                // We write out blank chars because we don't know how many cells will be rendered by the terminal
+                // then we draw the complex glyph on top of the blank chars.
+                WriteText(new string(' ', pixel.Width));
+                SetCaretPosition(position);
+                WriteText(pixel.Foreground.Symbol.Complex);
```

Comment:
A understand and appreciate this approach. Just as an idea: would not be better to keep a dictionary of probes (similar to what you did in Symbol)? Once we see no probe - we flush that complex glyph and same time reading the position - this way any next time we could already re-use the real width and do 0 additional instructions.